resources/ext/cli/config-migration

```diff
@@ -0,0 +1,243 @@
+#!/bin/bash
+
+migration_command="puppetdb config-migration"
+
+#############
+# FUNCTIONS #
+#############
+
+# Backs up a file, if it hasn't already been backed up
+#
+# $1 - file to backup
+#
+# Example:
+#
+#   backupfile "/etc/myconfig"
+#
+function backupfile() {
```

Comment:
As an alternative, I wonder if you could set backup_suffix=$(...date...) the the top of the script so that the current invocation would have a consistent suffix, and then just rely on the filesystem itself, instead of the state array  i.e. "if ! test -e "$target"; ...".